dockstore-common/src/main/java/io/dockstore/common/yaml/DockstoreYamlHelper.java

```diff
@@ -177,7 +177,21 @@ private static DockstoreYaml12 readDockstoreYaml12(final String content) throws
             final Yaml yaml = new Yaml(constructor);
             return yaml.load(content);
         } catch (Exception e) {
-            final String msg = ERROR_READING_DOCKSTORE_YML + e.getMessage();
+            String msg = ERROR_READING_DOCKSTORE_YML;
+            String errorMsg = e.getMessage();
+            final String cannotCreateProperty = "Unable to find property ";
+            if (errorMsg.contains(cannotCreateProperty)) {
+                String truncatedError = errorMsg.substring(errorMsg.indexOf(cannotCreateProperty) + cannotCreateProperty.length());
+                Pattern pattern = Pattern.compile("'(.+)'");
```

Comment:
1. Pattern.compile is relatively expensive. If the pattern isn't changing, which it's not, you should just compile it once. Best [link](https://rules.sonarsource.com/java/tag/performance/RSPEC-4248) I could find off-hand. Realistically, it's more of an issue in a loop; one call here isn't going to make a noticeable difference to the end user. Still, little things add up and it's easy enough to fix.
2. Since you're using a regex here, suggest you just use one regex for everything, rather than a combo of `contains` and regex, e.g., something like (not tested, probably has syntax errors, and may not match your target string): `Pattern.compile("Unable to find property '(.+)'");`

I also encourage the use of final when applicable. And I thought Codacy at one point flagged those, but I guess it's not any more.